source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -326,10 +326,16 @@ void DnsCacheImpl::forceRefreshHosts() {
       primary_host.second->active_query_->cancel(
           Network::ActiveDnsQuery::CancelReason::QueryAbandoned);
       primary_host.second->active_query_ = nullptr;
-      primary_host.second->timeout_timer_->disableTimer();
+      if (Runtime::isRuntimeFeature("envoy.reloadable_features.dfp_disable_dns_query_timeout") &&
+          timeout_interval_.count() > 0) {
```

Comment:
if you flipped this mid-Envoy-run you could end up with inconsistent results here I think.  I'd be inclined to create the timer based on flag, and check existence at the 4 months but alternately you could make this a restart flag, or latch a boolean.

Reply:
Done. I updated the flag to use a restart flag.